### DISCUSSION TITLE
AIO uses postgres, but nextcloud documentation recommends mysql.  Why the disparity?

### DISCUSSION THREAD

**Discussion by Anaxagoras-bc** [nextcloud/all-in-one#3975] [Questions] [answered]
So my current docker based nextcloud install is using mariadb, and i'm thinking about converting over to the AIO.  I realized that the aio install uses postgresql.  I normally look up best practices when setting these things up, so I decided to look into it.

the Nextcloud documentation recommends mysql/mariadb
https://docs.nextcloud.com/server/latest/admin_manual/configuration_database/linux_database_configuration.html

and system requirements section says the same thing

https://docs.nextcloud.com/server/latest/admin_manual/installation/system_requirements.html

however, i found a post here where @szaimen says Postgres works better than mysql/maria
https://github.com/nextcloud/all-in-one/discussions/2202

So i guess my question is, why the discrepancy?  Ultimately the AIO uses postgres, so it seems that's the answer, is the documentation just out of date?

**Comment by szaimen** [comment, marked as answer]
The main reason is that the Nextcloud GmbH has more people being familiar with mariadb/mysql than postgresql. Still, postgresql is more advanced and works better and it is also supported, that is why it was chosen here.

**Reply by xalt7x** [reply]
@szaimen 
Is it possible for Nextcloud team to reconsider the recommendation of MySQL in favor of PostgreSQL?
I believe, it should reduce issues for newcomers (e.g., "[1213 Deadlock found...](https://github.com/nextcloud/server/issues/6899)").   
Some time ago DB conversion [was an option](https://blog.jeanbruenn.info/2022/11/25/nextcloud-migrate-to-postgresql/) but currently it's [broken/disabled](https://github.com/nextcloud/server/issues/45257).  
With [PR 48666](https://github.com/nextcloud/server/pull/48666) applied, conversion works fine on instance without installed applications. However, it seems like DB migration script still needs some improvements for apps like "Talk"
> The following tables will not be converted: oc_talk_commands"

Thanks!

**Comment by realaaa** [comment]
great question yes, I was also wondering why this difference, will be trying PgSQL at some stage 

good discussion here as well - https://help.nextcloud.com/t/mariadb-or-postgresql/155593
